app/controllers/api/v1/users_controller.rb

```diff
@@ -30,6 +31,8 @@ def create
 
   # PATCH/PUT /users/:slug
   def update
+    authorize @user
```

Comment:
instead use  `before_action :authorize_object,  only: %i[update destroy]`

and then define the private method:

```
def authorize_user
    authorize @user
end
```